server/src/test/java/org/elasticsearch/common/TimeBasedUUIDGeneratorTests.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.common;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+
+public class TimeBasedUUIDGeneratorTests extends ESTestCase {
```

Comment:
I have specific unit tests for both `TimeBasedUUIDGenerator` and `TimeBasedKOrderedUUIDGenerator`...both of them generate UUID based on time and share the same requirements...indeed the only difference between them is the way bytes are ordered.